test/spectests/spectests.cpp

```diff
@@ -103,7 +121,17 @@ class test_runner
 
                 const auto registered_name = cmd.at("as").get<std::string>();
 
-                registered_names[registered_name] = module_name;
+                if (module_name == UnnamedModule)
+                {
+                    // Assign a name to unnamed module to avoid it being overwritten by another
+                    // unnamed one. Let the name be equal to registered name.
+                    assert(instances.find(registered_name) == instances.end());
```

Comment:
Should this be an assert or a fail? This requires spectest to be compiled in debug mode to be effective.

Reply:
Well it would be a failure of the assumption made by the testing code... I guess we can change it to failure for easier debugging.

Reply:
Changed to failure.